packages/invoices/src/OutboundShipment/DetailView/modals/ItemDetailsModal.tsx

```diff
@@ -1,435 +1,124 @@
-      const batchRowIdx = newBatchRows.findIndex(({ id }) => batch.id === id);
-      const batchRow = newBatchRows[batchRowIdx];
-      if (!batchRow) return null;
-      if (toAllocate < 0) return null;
-
-      const availableUnits =
-        batchRow.availableNumberOfPacks * batchRow.packSize;
-      const unitsToAllocate = Math.min(toAllocate, availableUnits);
-      const allocatedNumberOfPacks = Math.ceil(
-        unitsToAllocate / batchRow.packSize
-      );
-
-      toAllocate -= allocatedNumberOfPacks * batchRow.packSize;
-
-      newBatchRows[batchRowIdx] = {
-        ...batchRow,
-        numberOfPacks: allocatedNumberOfPacks,
-      };
-    });
-
-    // if over-allocated due to pack sizes available, reduce allocation as needed
-    if (toAllocate < 0) {
-      toAllocate *= -1;
-      validBatches.sort(sortByExpiryDesc).forEach(batch => {
-        const batchRowIdx = newBatchRows.findIndex(({ id }) => batch.id === id);
-        const batchRow = newBatchRows[batchRowIdx];
-        if (!batchRow) return null;
-        if (batchRow.packSize > toAllocate) return null;
-        if (batchRow.numberOfPacks === 0) return null;
-
-        const allocatedUnits = batchRow.numberOfPacks * batchRow.packSize;
-        const unitsToReduce = Math.min(toAllocate, allocatedUnits);
-        const numberOfPacks = Math.floor(
-          (allocatedUnits - unitsToReduce) / batchRow.packSize
-        );
-
-        toAllocate -= unitsToReduce;
-
-        newBatchRows[batchRowIdx] = {
-          ...batchRow,
-          numberOfPacks: numberOfPacks,
-        };
-      });
-    }
-
-    if (draft.status === InvoiceNodeStatus.New) {
-      const placeholderIdx = newBatchRows.findIndex(
-        ({ id }) => id === 'placeholder'
-      );
-      const placeholder = newBatchRows[placeholderIdx];
-
-      if (!placeholder) throw new Error('No placeholder within item editing');
-
-      newBatchRows[placeholderIdx] = {
-        ...placeholder,
-        numberOfPacks:
-          placeholder.numberOfPacks + toAllocate * (issuePackSize || 1),
-      };
-    }
-
-    setBatchRows(newBatchRows);
-  };
-
-  const onChangeRowQuantity = (batchId: string, value: number) => {
-    setBatchRows(issueStock(batchRows, batchId, value));
-  };
-
-  React.useEffect(() => {
-    if (isOpen) showDialog();
-    else {
-      onChangeItem(null);
-      hideDialog();
-    }
-  }, [isOpen]);
+  const onAllocate = allocateQuantities(
+    status,
+    draftOutboundLines,
+    setDraftOutboundLines
+  );
 
   return (
     <Modal
-      title={t(isEditMode ? 'heading.edit-item' : 'heading.add-item')}
-      cancelButton={<DialogButton variant="cancel" onClick={onCancel} />}
+      title={t(
+        mode === ModalMode.Update ? 'heading.edit-item' : 'heading.add-item'
+      )}
+      cancelButton={<DialogButton variant="cancel" onClick={onClose} />}
       nextButton={
         <DialogButton
-          disabled={isEditMode && isOnlyItem}
+          disabled={mode === ModalMode.Create}
```

Comment:
how about extending this to disable if this is the last or only item? 
or changing the `onNext` behaviour so that the next loops around? currently when I reach the last item, the button is enabled still, but does nothing when I click it

Reply:
Ah yes, I forgot about this!